receiver/prometheusreceiver/metrics_receiver_test.go

```diff
@@ -1400,6 +1406,22 @@ func testEndToEnd(t *testing.T, targets []*testData, useStartTimeMetric bool) {
 			target.validateFunc(t, target, results[target.name])
 		})
 	}
+
+	// verify state after shutdown is called
+	assert.Lenf(t, flattenTargets(rcvr.scrapeManager.TargetsAll()), 3, "expected 3 targets to be running")
+	shutdownCalled = true
+	require.NoError(t, rcvr.Shutdown(context.Background()))
+
+	assert.Len(t, flattenTargets(rcvr.scrapeManager.TargetsAll()), 0, "expected scrape manager to have no targets")
```

Comment:
should we just put this in the cleanup function, and remove the Shutdown call here?

Reply:
Yeah that's much cleaner, wasn't sure if it was ok to put assertions in the Cleanup handler but seems to work fine.